packages/token/src/erc20/erc20.cairo

```diff
@@ -288,6 +296,104 @@ pub mod ERC20Component {
         }
     }
 
+    /// The ERC20Permit trait implements the EIP-2612 standard, facilitating token approvals via
+    /// off-chain signatures. This approach allows token holders to delegate their approval to spend
+    /// tokens without executing an on-chain transaction, reducing gas costs and enhancing
+    /// usability.
+    /// The message signed and the signature must follow the SNIP-12 standard for hashing and
+    /// signing typed structured data.
+    ///
+    /// To safeguard against replay attacks and ensure the uniqueness of each approval via `permit`,
+    /// the data signed includes:
+    ///   - The address of the owner
+    ///   - The parameters specified in the `approve` function (spender and amount)
+    ///   - The address of the token contract itself
+    ///   - A nonce, which must be unique for each operation, incrementing after each use to prevent
+    ///   reuse of the signature - The chain ID, which protects against cross-chain replay attacks
+    ///
+    /// EIP-2612: https://eips.ethereum.org/EIPS/eip-2612
+    /// SNIP-12:  https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md
+    #[embeddable_as(ERC20PermitImpl)]
+    impl ERC20Permit<
+        TContractState,
+        impl ERC20: HasComponent<TContractState>,
+        +ERC20HooksTrait<TContractState>,
+        impl Nonces: NoncesComponent::HasComponent<TContractState>,
+        impl Metadata: SNIP12Metadata,
+        +Drop<TContractState>
+    > of interface::IERC20Permit<ComponentState<TContractState>> {
+        /// Sets the allowance of the `spender` over `owner`'s tokens after validating the signature
+        /// generated off-chain and signed by the `owner`.
+        ///
+        /// Requirements:
+        ///
+        /// - `owner` is a deployed account contract.
+        /// - `spender` is not the zero address.
+        /// - `deadline` is not a timestamp in the past.
+        /// - `signature` is a valid signature that can be validated with a call to `owner` account.
+        /// - `signature` must use the current nonce of the `owner`.
+        ///
+        /// Emits an `Approval` event.
+        fn permit(
+            ref self: ComponentState<TContractState>,
+            owner: ContractAddress,
+            spender: ContractAddress,
+            amount: u256,
+            deadline: u64,
+            signature: Array<felt252>
```

Comment:
We should favor Span over Array, but in this case we need Array (to make the `is_valid_signature` call to the owner's account, which expects a signature as `Array<felt252>`)

Reply:
Similar to the implementation in SRC9, we should receive the Span, and call into when calling `is_valid_signature`, since we only keep that Array for BC with the interface Id, but we should not continue adding more Array<felt252> when not needed. It would be nice if you could profile the calls using Span and into vs and Array to compare steps.

Reply:
Changed the input type to Span